cosmoz-data-nav.js

```diff
@@ -730,17 +723,29 @@
 				return;
 			}
 
+			const selected = this.selected;
```

Comment:
The selected element should be moved to first position in the Array. I don't see why sort would be used instead of simply removing it from where it is and putting it to first position.

Reply:
Well mostly I think it hurts readability, and extensibility.

There are now 7 lines to read and understand as opposed to one. 
You can't skip reading the block because it's not clear what it does until you've read it all.
You would also slow down to evaluate why selected = 0 or selectedIndex = 0 wouldn't apply for this code.
It would also fail on corner-cases like selected = 0 but not first task in queue.

I'm not sure this is more performant, and if it is, it wouldn't matter for a Number array of 3-10 items.

It will be messy to extend, if we, for example, want to sort the queue on more than one dimension. Selected, needs resize, etc, etc.

Reply:
Will revert this to sort.

Reply:
@nomego Reverted to sort.